### ISSUE TITLE
Check whether x/y varies makes sense for 3D plots

### ISSUE THREAD

**Issue by egeerardyn** [matlab2tikz/matlab2tikz#635]
Instead of #629, we should try if we can come up with some heuristics or code to determine whether it makes sense to put `x varies`(default) or `y varies` in 3D plots. 

The difference between both (as shown in #629, from where I've shamelessly stolen the graphs) is huge and obviously data dependent.
![pre_fix](https://cloud.githubusercontent.com/assets/2853795/7316205/ad64660c-ea7d-11e4-9b32-d56b2e65b414.png)
vs
![post_fix](https://cloud.githubusercontent.com/assets/2853795/7316213/c0b0c408-ea7d-11e4-862b-e7e047a96680.png)

However, I'm unsure whether doing the proper checks is easy (and fast). If so, I think it's better to have some common sense heuristics that at least fixes something like #629 and maybe some other "reasonable" cases.

**Comment by PeterPablo** [comment]
Before taking action, I personally would like to see an example that produces this issue with Matlab code.

**Comment by egeerardyn** [comment]
@PeterPablo You are right. It's probably best to even introduce a test case for this. I imagine two subplots where the left one benefits from `x varies`, the right one from `y varies`.

**Comment by PeterPablo** [comment]
Exactly. I am not yet convinced that `m2t` is misbehaving.

**Comment by egeerardyn** [comment]
Well, I'm certain it is not `m2t` that is misbehaving in those plots. But it's rather a limitation of `pgfplots`/`TikZ` and how it renders 3D plots.

But, as `pgfplots` has the option to work around this problem, we can at least try if we can make use of it.
